RHIDP-12593: Support pluginDivisionMode: schema for Single Database Deployments (#2099)
redhat-developer/red-hat-developers-documentation-rhdh: modules/configure_configuring-rhdh/proc-enable-plug-in-division-mode-schema.adoc

:_mod-docs-content-type: PROCEDURE

[id="enable-plug-in-division-mode-schema_{context}"]

> = Enable plug-in division mode schema 

Reviewer: ```suggestion
= Configure schema-based plugin isolation
```

Author: done

[role="_abstract"]

> Configure the `backend.database` block in `app-config.local.yaml` to enable the plug-in division mode for database schema division.

Reviewer: ```suggestion
If your database user is not granted `CREATEDB` privileges, for example on managed database services, configure schema division mode so that all plugins share a single database and each plugin uses its own PostgreSQL schema instead of a separate database.
```

Author: updated using gem tool and discussed with Fortune in slack. done.

[NOTE]
====

> By default, {product-very-short} automatically creates the required schemas. If a database user lacks `Create Schema` privileges, add `ensureSchemaExists: false` to the database configuration. The database administrator must create all required schemas.

Reviewer: ```suggestion
By default, when `pluginDivisionMode` is set to `schema`, {product-very-short} automatically creates the required schemas because the `ensureExists` configuration defaults to true. The database user must have `CREATE SCHEMA` privileges on the target database.

If your database user lacks `CREATE SCHEMA` privileges, set both `ensureExists: false` and `ensureSchemaExists: false` in the` backend.database` configuration. Your database administrator must create all required plugin schemas in the target database before starting {product-very-short}.
```

Author: done

====

.Procedure

. Open the OpenShift web console and navigate to the *ConfigMaps* section in the Red Hat Developer Hub project.

. Edit the `app-config.yaml` file.

. Navigate to the `backend.database` section and set the `pluginDivisionMode` parameter to `schema`.

> . Ensure the `client` is set to `pg` and verify the connection details for your single database:

Reviewer: ```suggestion
. Add pluginDivisionMode: schema to the backend.database section of your {product-very-short} configuration:
```

Author: done
